pkg/scheduler/plugins/loadaware/load_aware.go

```diff
@@ -146,75 +145,52 @@ func (p *Plugin) Filter(ctx context.Context, state *framework.CycleState, pod *c
-		if !status.IsSuccess() {
-			return status
-		}
+	prodPod := len(filterProfile.ProdUsageThresholds) > 0 && extension.GetPodPriorityClassWithDefault(pod) == extension.PriorityProd
+
+	var nodeUsage *slov1alpha1.ResourceMap
+	var usageThresholds map[corev1.ResourceName]int64
+	if prodPod {
+		usageThresholds = filterProfile.ProdUsageThresholds
 	} else {
-		var usageThresholds map[corev1.ResourceName]int64
+		if nodeMetric.Status.NodeMetric != nil {
+			if filterProfile.AggregatedUsage != nil {
+				nodeUsage = getTargetAggregatedUsage(
+					nodeMetric,
+					filterProfile.AggregatedUsage.UsageAggregatedDuration,
+					filterProfile.AggregatedUsage.UsageAggregationType,
+				)
+			} else {
+				nodeUsage = &nodeMetric.Status.NodeMetric.NodeUsage
+			}
+		}
 		if filterProfile.AggregatedUsage != nil {
 			usageThresholds = filterProfile.AggregatedUsage.UsageThresholds
 		} else {
 			usageThresholds = filterProfile.UsageThresholds
 		}
-		if len(usageThresholds) > 0 {
-			status := p.filterNodeUsage(node, nodeMetric, filterProfile)
-			if !status.IsSuccess() {
-				return status
-			}
-		}
 	}
-
-	return nil
-}
-
-func (p *Plugin) filterNodeUsage(node *corev1.Node, nodeMetric *slov1alpha1.NodeMetric, filterProfile *usageThresholdsFilterProfile) *framework.Status {
-	if nodeMetric.Status.NodeMetric == nil {
+	estimatedUsed, err := p.GetEstimatedUsed(node.Name, nodeMetric, pod, nodeUsage, prodPod)
+	if err != nil {
 		return nil
 	}
 
-	var usageThresholds map[corev1.ResourceName]int64
-	if filterProfile.AggregatedUsage != nil {
-		usageThresholds = filterProfile.AggregatedUsage.UsageThresholds
-	} else {
-		usageThresholds = filterProfile.UsageThresholds
-	}
-
-	for resourceName, threshold := range usageThresholds {
-		if threshold == 0 {
+	for resourceName, value := range usageThresholds {
+		if value == 0 {
 			continue
 		}
-		allocatable, err := p.estimator.EstimateNode(node)
-		if err != nil {
-			klog.ErrorS(err, "Failed to EstimateNode", "node", node.Name)
-			return nil
-		}
-		total := allocatable[resourceName]
-		if total.IsZero() {
-			continue
-		}
-		// TODO(joseph): maybe we should estimate the Pod that just be scheduled that have not reported
-		var nodeUsage *slov1alpha1.ResourceMap
-		if filterProfile.AggregatedUsage != nil {
-			nodeUsage = getTargetAggregatedUsage(
-				nodeMetric,
-				filterProfile.AggregatedUsage.UsageAggregatedDuration,
-				filterProfile.AggregatedUsage.UsageAggregationType,
-			)
-		} else {
-			nodeUsage = &nodeMetric.Status.NodeMetric.NodeUsage
-		}
-		if nodeUsage == nil {
+		total := getResourceValue(resourceName, allocatable[resourceName])
+		if total == 0 {
 			continue
 		}
-
-		used := nodeUsage.ResourceList[resourceName]
-		usage := int64(math.Round(float64(used.MilliValue()) / float64(total.MilliValue()) * 100))
-		if usage >= threshold {
+		usage := int64(math.Round(float64(estimatedUsed[resourceName]) / float64(total) * 100))
+		if usage > value {
```

Comment:
Also a cyclomatic complexity problem. Same above.

Reply:
I think it is necessary here.